dts/bindings/clock/fixed-clock.yaml

```diff
@@ -0,0 +1,37 @@
+#
+# Copyright (c) 2019 Derek Hageman <hageman@inthat.cloud>
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+---
+title: Generic fixed rate clock provider
+version: 0.1
+
+description: >
+    This is a representation of a generic fixed rate clock provider.
+
+properties:
+    compatible:
+      type: string
+      category: required
+      description: compatible strings
+      constraint: "fixed-clock"
+
+    label:
+      type: string
+      category: required
+      description: Human readable string describing the device (used by Zephyr for API name)
+      generation: define
+
+    clock-frequency:
+      type: int
+      description: output clock frequency (Hz)
+      generation: define
+      category: required
+
+    clocks:
```

Comment:
Is this correct, seems like there shouldn't be a 'clocks' property here.

Reply:
This is used in #15585 for clocks that use another clock as a reference.  e.g. SAM0 DFLL, which has a fixed 48 MHz frequency but can run in closed loop mode multiplying another clock to 48 MHz.

Reply:
Need to understand this more, but seems like this isn't a `fixed-clock` than.